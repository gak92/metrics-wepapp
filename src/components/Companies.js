import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useSearchParams } from 'react-router-dom';
import { getCompanies } from '../redux/actions';
import Company from './Company';

const Companies = () => {
  const dispatch = useDispatch();
  const companiesList = useSelector((state) => state.companiesReducer);
  const [searchParams, setSearchParams] = useSearchParams();

  useEffect(() => {
    // console.log(companiesList.length);
    if (companiesList.length === 0) {
      dispatch(getCompanies());
    }
  }, []);

  const searchHandler = (event) => {
    // console.log(event.target.value);
    const searchVal = event.target.value;
    if (searchVal) {
      setSearchParams({ searchVal });
    } else {
      setSearchParams({});
    }
  };

  return (
    <div>
      <input
        type="text"
        placeholder="Search company by Name ..."
        value={searchParams.keyword}
        onChange={searchHandler}
      />
      <div className="d-flex flex-wrap gap-1 justify-content-center">
        {
          companiesList.filter((company) => {
            const companyName = company.name.toLowerCase();
            const searchQuery = searchParams.get('searchVal');
            // console.log('Search Query: ', !searchQuery);
            if (searchQuery) {
              // console.log(companyName.includes(searchQuery));
              return companyName.includes(searchQuery);
            }
            // console.log('All true');
            return true;
          })
            .map((company) => (
              <Company
                key={company.symbol}
                symbol={company.symbol}
                name={company.name}
                price={company.price}
              />
            ))
        }
      </div>
    </div>
  );
};

export default Companies;
