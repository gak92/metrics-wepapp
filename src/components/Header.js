import React from 'react';

const Header = () => {
  const heading = 'Stock Companies';
  return (
    <div>
      {heading}
    </div>
  );
};

export default Header;
