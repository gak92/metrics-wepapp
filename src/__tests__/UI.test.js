import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import Companies from '../components/Companies';
import Company from '../components/Company';
import store from '../redux/configureStore';

describe('Testing UI', () => {
  it('Headers render correctly', () => {
    const tree = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Header contain text', () => {
    render(
      <Router>
        <Header />
      </Router>,
    );

    const element = screen.getByText('Financial Stock Companies');
    expect(element).toBeInTheDocument();
  });

  it('Test Companies', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <Companies />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Test Company', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <Company />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
