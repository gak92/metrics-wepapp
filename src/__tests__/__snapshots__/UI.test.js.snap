// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UI Headers render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="header"
      >
        Financial Stock Companies
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1
      class="header"
    >
      Financial Stock Companies
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing UI Test Companies 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <input
          class="mb-2 w-100"
          placeholder="Search company by Name ..."
          type="text"
          value=""
        />
        <div
          class="d-flex flex-wrap gap-1 justify-content-center"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <input
        class="mb-2 w-100"
        placeholder="Search company by Name ..."
        type="text"
        value=""
      />
      <div
        class="d-flex flex-wrap gap-1 justify-content-center"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing UI Test Company 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-dark text-white overflow-hidden card"
        style="width: 32%;"
      >
        <a
          class=""
          href="/undefined"
        >
          <img
            alt="Card image"
            class="card-img"
            src="stock_bg.jpg"
            style="opacity: 0.4;"
          />
          <div
            class="card-img-overlay"
          >
            <div
              class="d-flex justify-content-between card-title h5"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-right-long "
                data-icon="right-long"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M334.5 414c8.8 3.8 19 2 26-4.6l144-136c4.8-4.5 7.5-10.8 7.5-17.4s-2.7-12.9-7.5-17.4l-144-136c-7-6.6-17.2-8.4-26-4.6s-14.5 12.5-14.5 22l0 88L32 208c-17.7 0-32 14.3-32 32l0 32c0 17.7 14.3 32 32 32l288 0 0 88c0 9.6 5.7 18.2 14.5 22z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <p
              class="card-text"
            />
            <p
              class="card-text"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-dark text-white overflow-hidden card"
      style="width: 32%;"
    >
      <a
        class=""
        href="/undefined"
      >
        <img
          alt="Card image"
          class="card-img"
          src="stock_bg.jpg"
          style="opacity: 0.4;"
        />
        <div
          class="card-img-overlay"
        >
          <div
            class="d-flex justify-content-between card-title h5"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-right-long "
              data-icon="right-long"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M334.5 414c8.8 3.8 19 2 26-4.6l144-136c4.8-4.5 7.5-10.8 7.5-17.4s-2.7-12.9-7.5-17.4l-144-136c-7-6.6-17.2-8.4-26-4.6s-14.5 12.5-14.5 22l0 88L32 208c-17.7 0-32 14.3-32 32l0 32c0 17.7 14.3 32 32 32l288 0 0 88c0 9.6 5.7 18.2 14.5 22z"
                fill="currentColor"
              />
            </svg>
          </div>
          <p
            class="card-text"
          />
          <p
            class="card-text"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
